core/classifiers/CodeClassifier/CodeClassifier.py

```diff
@@ -543,7 +543,8 @@ def to_dict(self):
         """
         fingerprint_data = {
             unicode(index): {
-                u'fingerprint': self.training_data[index]['fingerprint'],
+                u'fingerprint': [
+                    list(f) for f in self.training_data[index]['fingerprint']],
```

Comment:
Not sure I understand what's going on here -- seems unintuitive. Maybe add an explanation?

Seems like you're creating a bunch of 1-element lists and I'm not sure why.

Reply:
The fingerprint consist of fingerprint components where each component is a tuple of length 2. The first element of the tuple is the component and the second is the position in source document from where first component is extracted. Both combined togather and combined with other components extracted (extraction according to winnowing algorithm) similarly from source document  make up the fingerprint of document.

Reply:
Oh, I get it now! Thanks for the explanation. Can you add what you just said to the docstring on line 506? In particular there's no element of a tuple in the docstring so that's why I was getting quite confused.

Reply:
Added the comment.